### DISCUSSION TITLE
too many curl processes make OS broken

### DISCUSSION THREAD

**Discussion by FulanXisen** [neovim/neovim#29019] [Q&A] [answered]
### Problem

I can't ensure the root reason of this problem. It happens when I use alacritty+zellij+neovim+lazynvim to do rust programming.

Their versions are: neovim=v0.10.0, lazynvim=master, plugins= Enabled: (6)coding.codeium, coding.yanky, editor.trouble-v3 , lang.rust, lang.toml, lsp.none-ls

When it happens, there are some abnormal things :
1. too many error in nvim ui: Error to spawn curl
2. shell unusable, temporal unaccess resouce
3. too many curl process in system monitor, about 4500+ curl processes.
4. most applications can't work, e.g. chrome can't search and open page, iterm can't be opened.

in short, OS was almost unusable. 

I get out of this situnation by kill some curl processes via system monitor so that I can open a shell, and then kill all curl processes. 

Does any know who cause this problem, or how to locate it?

### Steps to reproduce

using alacritty+zellij+neovim+lazynvim to do rust programming.

### Expected behavior

wait some minutes, there are so many curl processes

### Neovim version (nvim -v)

0.10.0 release

### Vim (not Nvim) behaves the same?

no vim

### Operating system/version

macOS 14,5

### Terminal name/version

alacritty 0.13.2

### $TERM environment variable

xterm-256color

### Installation

brew

**Comment by clason** [comment]
Sorry, without a minimal reproducer we can test with `nvim --clean -u test.lua` (preferably without plugins, because the default assumption here is that this is _their_ issue), we can't do anything.

**Reply by FulanXisen** [reply, marked as answer]
I just found out tha these curl processes are spawned by plenary.nvim. I think I should ask in its issues. 

Thanks for  your quick comment.
